src/providers/tfc/workspaceProvider.ts

```diff
@@ -86,28 +96,40 @@ export class WorkspaceTreeDataProvider implements vscode.TreeDataProvider<Worksp
       this.projectFilter = project.description;
       await vscode.commands.executeCommand('setContext', 'terraform.cloud.projectFilterUsed', true);
     }
+    this.cache = [];
     this.refresh();
     this.runDataProvider.refresh();
   }
 
-  getTreeItem(element: WorkspaceTreeItem): vscode.TreeItem | Thenable<vscode.TreeItem> {
+  getTreeItem(element: vscode.TreeItem): vscode.TreeItem | Thenable<vscode.TreeItem> {
     return element;
   }
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  getChildren(element?: any): vscode.ProviderResult<WorkspaceTreeItem[]> {
+  getChildren(element?: any): vscode.ProviderResult<vscode.TreeItem[]> {
     if (element) {
       return [element];
     }
 
+    return this.buildChildren();
+  }
+
+  private async buildChildren() {
     try {
-      return this.getWorkspaces();
+      this.cache = [...this.cache, ...(await this.getWorkspaces())];
```

Comment:
If I switch between organizations, I get a `Element with id '<long_id> is already registered`. I think we need to have the refresh call in the organization switcher somehow indicate we need to clear the cache?